freedomofpress/securedrop
# Adds new pubkey for Release Signing Key

## Status

Ready for review

## Description of Changes

Towards #5923 

Bumps the version of the `securedrop-keyring` package, preserving the
old/current release signing key, but adding a new pubkey.

  * Old/current fingerprint: 22245C81E3BAEB4138B36061310F561200F4AD77
  * New/next fingerprint: 2359E6538C0613E652955E6C188EDD3B7B22E6A3

This this is a soft rotation, we'll make sure that all instances have
the new key first, then later remove reference to the old key.

As with previous updates, here's the command I used to import the new
key into the keyring:

    gpg --no-default-keyring --keyring \
    install_files/securedrop-keyring/etc/apt/trusted.gpg.d/securedrop-keyring.gpg \
    --import \
    install_files/ansible-base/roles/install-fpf-repo/files/fpf-signing-key-2021.pub

Note the tweak to the target file to import, i.e. the "2021" suffix.

## Testing

1. Review the transition statement in https://github.com/freedomofpress/securedrop/issues/5923#issuecomment-833846114
2. Ensure that the new key introduced in this PR, i.e. `2359E6538C0613E652955E6C188EDD3B7B22E6A3`, was used to sign that statement.
3. Ensure that the current/old key, `22245C81E3BAEB4138B36061310F561200F4AD77`, was used to sign that statement.
4. Ensure that the keyring changes in this PR preserve the old pubkey while adding the new pubkey. 
5. Inspect the GUI updater logic, and confirm it's appropriate for the rotation plan. 

## Deployment
We intend to include these changes as part of 1.8.2: https://github.com/freedomofpress/securedrop/milestone/73 These changes should be reviewed in the context of ensuring continued updates for both servers and Journalist/Admin Workstations.
